src/chronicle_rsm.erl

```diff
@@ -141,9 +142,11 @@ sync(Name, Timeout) ->
 
 leader_request(Name, Request, TRef) ->
     Deadline = read_deadline(TRef),
-    case call(?SERVER(Name),
-              {leader_request, Request, Deadline}, leader_request,
-              infinity) of
+    Result = ?TIME(<<"leader_request">>, {ok, _},
```

Comment:
1. All Requests are logged against the same stat which are not going to have similar response times so this stat is not useful.
2. Implicitly only recording stats for leader_requests which have {ok, Reply} as the response?  {error,timeout} is skipped.
3. stats recording happens not on the leader but the node request originated on it looks like, so the stats will be distributed on nodes making requests. 

Could you help me understand what is the motivation and the intent here since adding this change doesn't help in the triage of MB-57128?

Reply:
1. Yes, only cm_chronicle_rsm_seconds_bucket{op="leader_request"} will exist with this change but we can add others. e.g. cm_chronicle_rsm_seconds_bucket{op="query"} or cm_chronicle_rsm_seconds_bucket{op="get_local_revision"} etc.  This stat will be useful in terms of providing an indication of how long the requests are taking in the different time buckets. 
2. The stat is recorded for all responses...which goes into "Result" and then "case Result of" is done.
3. Good point. It wasn't intended and "local" requests will skew results to the lower buckets but knowing the longer buckets will be useful.

IMO leader_request is a function in chronicle where knowing the latencies of the requests might lead to further actions such as finer grained stats, examination of code, etc.  Having stats is a first step in knowing whether or not futher steps are needed.